ophyd/utils/epics_pvs.py

```diff
@@ -325,3 +326,13 @@ def records_from_db(fn):
         ret.append((rtype, record))
 
     return ret
+
+def raise_if_disconnected(fcn):
+    '''Decorator to catch attempted access to disconnected EPICS channels.'''
+    @functools.wraps(fcn)
+    def wrapper(self, *args, **kwargs):
+        if self.connected:
+            return fcn(self, *args, **kwargs)
+        else:
+            raise Exception('{} is not connected'.format(self.name))
```

Comment:
I see what you mean. But yeah, unfortunately with just the exception class `Exception` you're left with no real choice but such ugliness. I've both seen and (regrettably) had to write code that checks the message like the above :expressionless: